esphomeyaml/cookbook/sonoff-fishpond-pump.rst

```diff
@@ -0,0 +1,367 @@
+    [11:13:27][C][wifi:241]:   IP Address: 192.168.13.15
+    [11:13:27][C][wifi:243]:   BSSID: xx:xx:xx:xx:xx:xx
+    [11:13:27][C][wifi:245]:   Hostname: 'fishpond'
+    [11:13:27][C][wifi:250]:   Signal strength: -91 dB ▂▄▆█
+    [11:13:27][C][wifi:251]:   Channel: 1
+    [11:13:27][C][wifi:252]:   Subnet: 255.255.255.0
+    [11:13:27][C][wifi:253]:   Gateway: 192.168.13.1
+    [11:13:27][C][wifi:254]:   DNS1: 192.168.13.1
+    [11:13:27][C][wifi:255]:   DNS2: 0.0.0.0
+    [11:13:27][C][binary_sensor.status:046]: Status Binary Sensor 'esp_fishpond_system_status'
+    [11:13:27][C][binary_sensor.status:046]:   Device Class: 'connectivity'
+    [11:13:28][C][switch.gpio:049]: GPIO Switch 'esp_fishpond_gpio12'
+    [11:13:28][C][switch.gpio:050]:   Pin: GPIO12 (Mode: OUTPUT)
+    [11:13:28][C][switch.gpio:066]:   Restore Mode: Restore (Default to OFF)
+    [11:13:28][C][binary_sensor.gpio:023]: GPIO Binary Sensor 'esp_fishpond_gpio0'
+    [11:13:28][C][binary_sensor.gpio:024]:   Pin: GPIO0 (Mode: INPUT, INVERTED)
+    [11:13:28][C][binary_sensor.gpio:023]: GPIO Binary Sensor 'esp_fishpond_gpio14'
+    [11:13:28][C][binary_sensor.gpio:024]:   Pin: GPIO14 (Mode: INPUT_PULLUP)
+    [11:13:28][C][output.esp8266_pwm:028]: ESP8266 PWM:
+    [11:13:28][C][output.esp8266_pwm:029]:   Pin: GPIO13 (Mode: OUTPUT)
+    [11:13:28][C][output.esp8266_pwm:030]:   Frequency: 1000.0 Hz
+    [11:13:28][C][logger:099]: Logger:
+    [11:13:28][C][logger:100]:   Level: DEBUG
+    [11:13:28][C][logger:101]:   Log Baud Rate: 115200
+    [11:13:28][C][light.state:266]: Light 'esp_fishpond_gpio13'
+    [11:13:28][C][light.state:268]:   Default Transition Length: 1000 ms
+    [11:13:28][C][light.state:269]:   Gamma Correct: 2.80
+    [11:13:28][C][switch.restart:034]: Restart Switch 'esp_fishpond_system_restart'
+    [11:13:28][C][switch.restart:034]:   Icon: 'mdi:restart'
+    [11:13:28][C][time.homeassistant:032]: Home Assistant Time:
+    [11:13:28][C][time.homeassistant:033]:   Timezone: 'SAST-2'
+    [11:13:28][C][sensor.wifi_signal:042]: WiFi Signal 'esp_fishpond_system_wifi_signal'
+    [11:13:28][C][sensor.wifi_signal:042]:   Unit of Measurement: 'dB'
+    [11:13:28](Message skipped because it was too big to fit in TCP buffer - This is only cosmetic)
+    [11:13:28](Message skipped because it was too big to fit in TCP buffer - This is only cosmetic)
+    [11:13:28][C][api:072]: API Server:
+    [11:13:28][C][api:073]:   Address: 192.168.13.15:6053
+    [11:13:28][C][ota:129]: Over-The-Air Updates:
+    [11:13:28][C][ota:130]:   Address: 192.168.13.15:8266
+
+
+
+2.4. Test now with OTA flashing
+*******************************
+
+Before installing the Sonoff, do a final OTA test, and this time selecting the OTA option and NOT the USB option when reflashing.
+
+.. code-block:: bash
+
+    esphomeyaml fishpond.yaml run
+
+Once these actions succeeded you are pretty much in the clear and can be sure your device is ready.
```

Comment:
I think maybe this could (at least partially) be moved to the [sonoff basic](https://esphomelib.com/esphomeyaml/devices/sonoff_basic.html) guide - Otherwise there'll be the same thing in multiple locations and that's a real mess to maintain. This document could then point to that guide for flashing the first firmware.

Reply:
The PR looks good to me at the current state. Do you want to move the programming instructions to the [sonoff basic](https://esphomelib.com/esphomeyaml/devices/sonoff_basic.html) guide?

Also, I cannot find a reference to "my previous comment about cookbook vs tutorial comes in". could you elaborate?

Reply:
Greetings, 

> The PR looks good to me at the current state. Do you want to move the programming instructions to the sonoff basic guide?

I have made the changes as I best I could see fit, moved documentation into a subheading and added :doc links (please check and fix if needed, .rst renders them not as links)

> Also, I cannot find a reference to "my previous comment about cookbook vs tutorial comes in". could you elaborate?

On my first attempts when contributing, I added all my text to the changes to the git comment/wysiwyg area and clicked "Resolve Conversation"... that does not then add the text to the conversation. That is where it got lost.

I was just making an observation I was not sure what a 'cookbook' must contain and some of the content in my initial submission felt very much tutorial like. This has been addressed and content has been moved now to the [sonoff basic guide](https://esphomelib.com/esphomeyaml/devices/sonoff_basic.html)